### PR TITLE
add example for comparing subscriptions from different providers

### PR DESCRIPTION
**Pull request by 0xlosh**
Closes: https://github.com/alloy-rs/examples/issues/146

added new_heads to compare, simply run `cargo run --example new_heads -- -r reth:ws://localhost:8546 -r alchemy:xxx -r other_provider:other_url -r ...` to start

### CODE REVIEW

File: examples/comparison/examples/new_heads.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let cli = Cli::parse();
+    let rpcs: Vec<(&str, &str)> = cli.rpcs.iter().filter_map(|s| s.split_once(':')).collect();
+
+    let (sx, mut rx) = mpsc::unbounded_channel();
+    for (name, url) in rpcs {
+        let sx = sx.clone();
+        let name = name.to_string();
```
Comment (mattsse) [inline examples/comparison/examples/new_heads.rs:27]: you can convert this to `Arc<str>` then this is cheaper to clone

File: examples/comparison/examples/new_heads.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            let mut stream = match provider.subscribe_blocks().await {
+                Ok(stream) => stream.into_stream(),
+                Err(e) => {
+                    eprintln!("skipping {} at {} because of error: {}", name, url, e);
+                    return;
+                }
+            };
+            while let Some(block) = stream.next().await {
+                if let Err(e) =
+                    sx.send((name.clone(), block.header.number, Utc::now().timestamp_millis()))
```
Comment (mattsse) [inline examples/comparison/examples/new_heads.rs:47]: lets send the entire block and Utc::now

**Review by mattsse — CHANGES_REQUESTED** [review]
tysm, this looks pretty good, only have some nits

**Comment by 0xlosh** [comment]
> tysm, this looks pretty good, only have some nits

will edit tmw and add pending_txs too, im not sure what gakonst meant when he mentioned to test `eth_getBlock` against other providers too, what exactly is there to compare?

File: examples/comparison/examples/new_heads.rs (hunk 2 above)
Comment (0xlosh) [reply examples/comparison/examples/new_heads.rs:47]: i think using Utc::now() is no precise enough, the diff ist always gonna be 0s

Comment (mattsse) [reply examples/comparison/examples/new_heads.rs:47]: I see, perhaps we need to track relative timestamps instead

**Comment by 0xlosh** [comment]
`cargo run --example new_heads -- -r reth:ws://localhost:8546 -r alchemy:wss://eth-mainnet.g.alchemy.com/v2/<api_key>`
Screenshot 2024-10-05 at 20 01 05

`cargo run --example pending_txs -- -r reth:ws://localhost:8546 -r alchemy:wss://eth-mainnet.g.alchemy.com/v2/<api_key>`
Screenshot 2024-10-05 at 20 01 23

unfortunately i dont have a local geth node running along to test that too, feel free to edit the code

**Comment by mattsse** [comment]
@0xlosh here's a reth node  ws://69.67.151.138:8546 

please reply once you saved this, so I can delete this
feel free to keep using this :)

**Comment by 0xlosh** [comment]
@mattsse thanks got it, but im running my own reth node actually

**Review by mattsse — APPROVED** [review]

**Review by zerosnacks — APPROVED** [review]
Thanks @0xlosh! LGTM

**Pull request merged**